library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -408,14 +410,19 @@ applyTerm direction pref trm = do
                 Nothing -> do
                     simplified <-
                         if isConcrete t && isJust config.llvmApi && attributes.canBeEvaluated
-                            then do
-                                -- LLVM simplification proceeds top-down and cuts the descent
-                                -- FIXME run this in IO
-                                let result = simplifyTerm (fromJust config.llvmApi) config.definition t (sortOfTerm t)
-                                when (result /= t) $ do
-                                    setChanged
-                                    traceRuleApplication t Nothing (Just "LLVM") Nothing $ Success result
-                                pure result
+                            then -- LLVM simplification proceeds top-down and cuts the descent
+                            -- FIXME run this in IO
+
+                                simplifyTerm (fromJust config.llvmApi) config.definition t (sortOfTerm t)
+                                    >>= \case
+                                        Left (LLVM.LlvmError err) -> do
+                                            logWarn $ decodeUtf8 err
+                                            apply config t
```

Comment:
should we fall back on equations or return original `t` here?

Reply:
Since simplifications are "optional", as in we try to apply them but may fail, I suggest we indeed emit a warning and return the original term. 

I think we need to push from the use-case here: we attempted to apply a simplification and got an exception. In Petar's use-case, as I understand, the simplification would not apply --- we should go on trying other ones that match. @PetarMax does that sound reasonable?